File: frontend/src/types/runs.ts
import { RunConfig } from './config';

export interface RunResult {
  run_id: string;
  timestamp: string;
  config: RunConfig;
  processed_files: number;
  matched_count: number;
  unmatched_count: number;
  log_file: string;
  plots_dir: string;
  summary_xlsx: string | null;
}

export interface RunSummary {
  run_id: string;
  timestamp: string;
  profile_set_id: string;
  processed_files: number;
  matched_count: number;
  unmatched_count: number;
  dominance_delta: number;
  balance_threshold: number;
  window_secs: number;
}

File: backend/app/models/runs.py
from pydantic import BaseModel
from .config import RunConfig


class RunResult(BaseModel):
    run_id: str
    timestamp: str
    config: RunConfig
    processed_files: int
    matched_count: int
    unmatched_count: int
    log_file: str
    plots_dir: str
    summary_xlsx: str | None = None


class RunSummary(BaseModel):
    run_id: str
    timestamp: str
    profile_set_id: str
    processed_files: int
    matched_count: int
    unmatched_count: int
    dominance_delta: float
    balance_threshold: float
    window_secs: int
